packages/deployments/contracts/contracts_forge/PromiseRouter.t.sol

```diff
@@ -0,0 +1,349 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+import "./ForgeHelper.sol";
+
+import "../contracts/Connext.sol";
+import "../contracts/interfaces/ICallback.sol";
+import "../contracts/nomad-xapps/contracts/promise-router/PromiseRouter.sol";
+import {Home} from "../contracts/nomad-core/contracts/Home.sol";
+
+// running tests (with logging on failure):
+// yarn workspace @connext/nxtp-contracts test:forge -vvv
+// run a single test:
+// yarn workspace @connext/nxtp-contracts test:forge -m testAddRouterAlreadyApproved -vvv
+
+// other forge commands: yarn workspace @connext/nxtp-contracts forge <CMD>
+// see docs here: https://onbjerg.github.io/foundry-book/index.html
+
+contract MackPromiseRouter is PromiseRouter {
+  using TypedMemView for bytes;
+  using TypedMemView for bytes29;
+  using PromiseMessage for bytes29;
+
+  function mockHandle(
+    address callbackAddress,
+    bool returnSuccess,
+    bytes calldata returnData
+  ) public {
+    bytes32 transferId = "A";
+
+    bytes memory message = PromiseMessage.formatPromiseCallback(transferId, callbackAddress, returnSuccess, returnData);
+    bytes29 _msg = message.ref(0).mustBePromiseCallback();
+
+    promiseMessages[transferId] = message;
+  }
+}
+
+contract MockHome {
+  function dispatch(
+    uint32 _destinationDomain,
+    bytes32 _recipientAddress,
+    bytes memory _messageBody
+  ) external {
+    1 == 1;
+  }
+}
+
+contract MockCallback is ICallback {
+  function callback(
+    bytes32 transferId,
+    bool success,
+    bytes memory data
+  ) external {
+    require(data.length != 0);
+  }
+}
+
```

Comment:
It seems generally we should have a `Mock.sol` that includes mocks for all of the constituent contracts since we seem to define the same ones in-file.

would be good to start creating that file now, but shouldn't block the PR

Reply:
Created!